openstudiocore/src/model/ZoneHVACEquipmentList.cpp

```diff
@@ -438,7 +444,40 @@ boost::optional<double> ZoneHVACEquipmentList_Impl::sequentialCoolingFraction(co
       OS_ASSERT(wo);
 
       if (wo->handle() == equipment.handle()) {
-        result = group.getDouble(OS_ZoneHVAC_EquipmentListExtensibleFields::ZoneEquipmentSequentialCoolingFraction, true);
+        boost::optional<WorkspaceObject> wo = group.cast<WorkspaceExtensibleGroup>().getTarget(OS_ZoneHVAC_EquipmentListExtensibleFields::ZoneEquipmentSequentialCoolingFractionScheduleName);
+        schedule = wo->optionalCast<ScheduleConstant>();
```

Comment:
Refactor this function to use a bool on line 434, and deal with schedule directly in the block around 447, because Ideally we do the following:
* If a Schedule was found, and it's a ScheduleConstant => return that double
* If a Schedule was found, and it's NOT a ScheduleConstant => issue a LOG(Warn, "message")
* If no schedule was found, return boost::none

Same applies to the `sequentialHeatingFraction` getter.